### PR TITLE
ttx: use default network identity if not specified

### PR DESCRIPTION
**Pull request by adecaro**
This commit allows the developer to create a new
nomimal token transaction with the default identity
of the target network.
Comments have been added and the fungible integration test
now generate also nominal transactions to issue tokens.

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: integration/token/fungible/views/issue.go [hunk 1]
```diff
@@ -65,14 +67,25 @@ func (p *IssueCashView) Call(context view.Context) (interface{}, error) {
...
-		),
-	)
+	// The issuer creates a new token transaction and specify the auditor that must be contacted to approve the operation.
+	var tx *ttx.Transaction
+	if p.Anonymous {
+		tx, err = ttx.NewAnonymousTransaction(
+			context,
+			ttx.WithAuditor(
+				view2.GetIdentityProvider(context).Identity("auditor"), // Retrieve the auditor's FSC node identity
+			),
```
Comment (mbrandenburger) [inline integration/token/fungible/views/issue.go:77]: could you store the `WithAuditor` option in a var before and use it in the if/else clause?

Also, there might be an issue if `NewTransaction` or `NewAnonymousTransaction` returns an error and `tx` becomes `nil`. I suggest that you move the err check right after the if/else before setting metadata.

Comment (mbrandenburger) [reply integration/token/fungible/views/issue.go:77]: Actually, I am wondering why we have `NewAnonymousTransaction` and `NewTransaction`. All the parameters of these functions could be provided via options, can't we?

Comment (adecaro) [reply integration/token/fungible/views/issue.go:77]: true, but with a clean different API, it is difficult to get it wrong 😅

Comment (adecaro) [reply integration/token/fungible/views/issue.go:77]: I'll move the error check 👍

File: token/services/ttx/endorse.go [hunk 2]
```diff
@@ -367,6 +363,34 @@ func (c *collectEndorsementsView) requestApproval(context view.Context) (*networ
...
+		// am I the auditor?
+		if view2.GetSigService(context).IsMe(c.tx.Opts.Auditor) {
+			//
+		} else {
+			// connect to the remote auditor
+			_, err := context.RunView(newAuditingViewInitiator(c.tx))
+			if err != nil {
+				return nil, errors.WithMessagef(err, "failed requesting auditing from [%s]", c.tx.Opts.Auditor.String())
+			}
+		}
```
Comment (mbrandenburger) [inline token/services/ttx/endorse.go:377]: if not `!view2.GetSigService(context).IsMe(c.tx.Opts.Auditor)` and get rid of the `else`?

**Review by KElkhiyaoui — CHANGES_REQUESTED** [review]
minor changes are needed.

File: integration/token/fungible/views/issue.go [hunk 3]
```diff
@@ -65,17 +67,28 @@ func (p *IssueCashView) Call(context view.Context) (interface{}, error) {
...
 	// At this point, the issuer is ready to prepare the token transaction.
-	// The issuer creates an anonymous transaction (this means that the resulting Fabric transaction will be signed using idemix, for example),
-	// and specify the auditor that must be contacted to approve the operation
-	tx, err := ttx.NewAnonymousTransaction(
-		context,
-		ttx.WithAuditor(
-			view2.GetIdentityProvider(context).Identity("auditor"), // Retrieve the auditor's FSC node identity
-		),
-	)
+	// The issuer creates a new token transaction and specify the auditor that must be contacted to approve the operation.
```
[the remark this answers is not in the document]
Comment (adecaro) [reply integration/token/fungible/views/issue.go:70]: okay

File: token/services/ttx/endorse.go [hunk 4]
```diff
@@ -63,51 +63,47 @@ func (c *collectEndorsementsView) Call(context view.Context) (interface{}, error
...
 	}
 
-	// Cleanup
-	if !c.tx.Opts.Auditor.IsNone() {
-		session, err := context.GetSession(nil, c.tx.Opts.Auditor)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed getting auditor's session")
-		}
-		session.Close()
+	// Cleanup audit
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/services/ttx/endorse.go:104]: changed name to better reflect the meaning

**Comment by adecaro** [comment]
@KElkhiyaoui, @mbrandenburger, sorry, there were changes not related to issue addressed by this PR. I have cleaned up everything.

**Review by mbrandenburger — APPROVED** [review]
LGTM

**Review by KElkhiyaoui — APPROVED** [review]

**Pull request merged**